lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -149,12 +166,44 @@ define((require, exports, module) ->
       else
         callback(null, @cache)
 
+  ###
+  * Show popup if token type at the current pos is one of the given array elements.
+  * @ (editor) --> editor
+  * @ (list of strings) -- allowedTypes
+  ###
+  showPopupIfTokenIsOneOfTypes = (editor, allowedTypes) ->
+    if editor.completer?
+      pos = editor.getCursorPosition()
+      session = editor.getSession()
+      token = editor.session.getTokenAt(pos.row, pos.column)
+
+      if token?
+        for type in allowedTypes
+          if LatexParsingContext.isType(token, type)
+            editor.completer.showPopup(editor)
+            break
+
   class TexCompleter
       constructor: ->
-        @refGetter = new ReferenceGetter()
-      @init: (editor) ->  init(editor,  {win: "enter", mac: "enter"})
-
+        @refCache = new CompletionsCache(processReferenceJson)
+        @citeCache = new CompletionsCache(processCitationJson)
+      @init: (editor) ->
+        init(editor,  {win: "enter", mac: "enter"})
```

Comment:
move that init closer to the class definition, or maybe just move the code into this init